src/python/pants/backend/python/target_types.py

```diff
@@ -189,6 +190,21 @@ class PexPlatformsField(StringOrStringSequenceField):
 
     alias = "platforms"
 
+    @classmethod
+    def compute_value(
+        cls, raw_value: Optional[Iterable[str]], *, address: Address
+    ) -> Optional[Tuple[str, ...]]:
+        if isinstance(raw_value, str) and address.is_base_target:
+            warn_or_error(
```

Comment:
Nit: the `if`+`warn_or_error` combo is what `deprecated_conditional` is meant to accomplish. It doesn't really have a great advantage over this usage, other than that it's a bit more obvious that the conditional/if exists only to serve the deprecation.

Reply:
Oh, true. I'll fix in a related followup.